collections/_article/dev-snapshot-godot-4-2-beta-1.md

```diff
@@ -0,0 +1,384 @@
+
+Regular and documentation comments are seeing some upgrades as well. Special TODO markers (such as `TODO`, `FIXME`, etc) are now highlighted in the code editor to better grab your attention ([GH-79761](https://github.com/godotengine/godot/pull/79761)). You can configure both keywords and colors that are given to them in the editor settings. Documentation comments now support `@deprecated` and `@experimental` tags to mark class members accordingly in the generated documentation ([GH-78941](https://github.com/godotengine/godot/pull/78941)). Documentation generation itself has also been improved ([GH-80745](https://github.com/godotengine/godot/pull/80745)). All these changes were submitted by, once again, [Danil Alexeev](https://github.com/dalexeev).
+
+![TODO and FIXME comments highlighted in the code editor](/storage/blog/godot-4-2-beta/todo-highlighting.webp)
+
+To close things off for GDScript, a contribution from [Ryan Brue](https://github.com/ryanabx) improves hovered symbol resolution, fixes various bugs related to renaming, and implements reference lookup, all in one go ([GH-80973](https://github.com/godotengine/godot/pull/80973)). Ryan also added a new command line argument, `--lsp-port` which is useful for authors of LSP plugins to support both Godot 3.x and 4.x, as well as multiple editor instances ([GH-81844](https://github.com/godotengine/godot/pull/81844)).
+
+
+### XR
+
+Godot's lead XR maintainer [Bastiaan Olij](https://github.com/BastiaanOlij) recently [published a progress report](/article/godot-xr-update-sep-2023) on new and upcoming XR features, which you should definitely read! Some highlights from the list of changes, implemented by Bastiaan himself, include foveated rendering support, ported from Godot 3.x ([GH-80881](https://github.com/godotengine/godot/pull/80881)), and access to raw hand-tracking data ([GH-78032](https://github.com/godotengine/godot/pull/78032)).
+
+On top of that, a significant contribution from [konczg](https://github.com/konczg) opens up the OpenXR module to be extended and interacted with from the GDExtension API ([GH-68259](https://github.com/godotengine/godot/pull/68259)). We can't wait for your projects utilizing this capability to its fullest!
+
+
+-----
+
+## Downloads
+
+The downloads for this pre-release build can be found in our GitHub repository:
+
+* [**Download Godot 4.2 beta 1**](https://github.com/godotengine/godot-builds/releases/tag/4.2-beta1).
+
+**Standard build** includes support for GDScript and GDExtension.
+
+**.NET build** (marked as `mono`) includes support for C#, as well as GDScript and GDExtension.
+- .NET build requires [.NET SDK 6.0](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) or [7.0](https://dotnet.microsoft.com/en-us/download/dotnet/7.0) installed in a standard location.
+- To export to Android, .NET 7.0 is required. To export to iOS, .NET 8.0 is required. Make sure to set the target framework in the `.csproj` file.
```

Comment:
I think for iOS the target framework is set automatically. Not sure about Android. Can you confirm @raulsntos?
```suggestion
- To export to Android, .NET 7.0 or later is required. To export to iOS, .NET 8.0 is required. Make sure to set the target framework in the `.csproj` file.
```

Reply:
I pushed the suggestion, but keeping this unresolved so Raul doesn't miss the message :)

Reply:
The right target is set for newly created projects (both Android and iOS). For existing projects, the user needs to edit the `.csproj` manually.

We were thinking about updating the `.csproj` automatically (see https://github.com/godotengine/godot/pull/82927) but automatically editing the `.csproj` is risky so I don't think this will be merged for 4.2.

Reply:
I saw the WIP documentation PR, it seems like we're adding it for new projects (with conditionals), but not for existing projects. So it makes sense to call it out here.